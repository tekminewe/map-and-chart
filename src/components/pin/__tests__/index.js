import React from 'react';
import { render, fireEvent, wait } from '@testing-library/react';
import Pin from '..';

describe('Pin', () => {
  test('should not show pop up by default', () => {
    const { container } = render(<Pin lng={123} lat={123} />);
    expect(container).toMatchSnapshot();
  });

  test('should toggle pop up when click on pin', () => {
    const { container, getByTestId } = render(<Pin lng={123} lat={123} />);

    const pin = getByTestId('pin');

    fireEvent.click(pin, {});
    expect(container).toMatchSnapshot();

    fireEvent.click(pin, {});
    expect(container).toMatchSnapshot();
  });

  test('should toggle pop up when click on popup', async () => {
    const { container, getByTestId } = render(<Pin lng={123} lat={123} />);

    const pin = getByTestId('pin');
    fireEvent.click(pin, {});

    const popup = getByTestId('pop_up');
    fireEvent.click(popup, {});
    expect(container).toMatchSnapshot();
  });
});
