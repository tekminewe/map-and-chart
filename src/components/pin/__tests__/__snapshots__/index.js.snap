// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pin should not show pop up by default 1`] = `
<div>
  <svg
    class="sc-AxjAm evlJpc"
    data-testid="pin"
  >
    pin.svg
  </svg>
</div>
`;

exports[`Pin should toggle pop up when click on pin 1`] = `
<div>
  <div
    class="sc-AxirZ fLvKCS"
    data-testid="pop_up"
  >
    <p>
      123
      , 
      123
    </p>
  </div>
  <svg
    class="sc-AxjAm evlJpc"
    data-testid="pin"
  >
    pin.svg
  </svg>
</div>
`;

exports[`Pin should toggle pop up when click on pin 2`] = `
<div>
  <svg
    class="sc-AxjAm evlJpc"
    data-testid="pin"
  >
    pin.svg
  </svg>
</div>
`;

exports[`Pin should toggle pop up when click on popup 1`] = `
<div>
  <svg
    class="sc-AxjAm evlJpc"
    data-testid="pin"
  >
    pin.svg
  </svg>
</div>
`;
